### PR TITLE
Fixes for continuous batching

### PR DESCRIPTION
**Pull request by remi-or**
Some architectures like `llama` alter the attention mask if it is not a tensor, which was not compatible with the way CB created and handled the attention mask. Now, arguments like `attention_mask`, `cumulative_seqlens_k` and `max_seqlen_k` are tensors or ints unless the model is hybrid, in which case they are dictonnaries. This is the main fix, but PR also:
- cleans up how those arguments are built, reset, and delivered to the model to make things tidier
- adds support for attention sink in `eager_paged`
- explicitly disables (for now) cuda graphs and sampling in CB
- adds test for CB that pass on MI325 and H100 (but it's a little sketchy for some models there)
- fixes some telemetry metrics that were broken when hybrid allocation were added

### CODE REVIEW

File: src/transformers/generation/continuous_batching/continuous_api.py [hunk 1]
```diff
@@ -147,97 +147,127 @@ def __init__(
...
+        for i in range(self.cache.num_groups):
+            self.write_index_storage[i][:t].fill_(-1)
+            self.read_index_storage[i][: t + c].fill_(-1)
 
     def get_model_kwargs(self) -> PagedAttentionArgs:
         """Get model keyword arguments for the current batch."""
         # Compute the slice to return
-        t = self.total_query_length if self.slice_inputs else self.write_index.size(-1)
+        t = self.total_query_length if self.slice_inputs else self.write_index_storage[0].size(-1)
         b = self.total_batch_size
```
Comment (ArthurZucker) [inline src/transformers/generation/continuous_batching/continuous_api.py]: in general let's try to avoid single letter variables, and batchsize might be a bad name! (it was already there but noticed)

File: src/transformers/generation/continuous_batching/continuous_api.py [hunk 2]
```diff
@@ -283,75 +313,79 @@ def _handle_request_error(self, error, state: RequestState):
...
             past_length = state.position_offset
-            query_length = len(next_input_ids)
-            key_length = query_length + past_length
+            query_length = len(state.prompt_ids)
+            seqlens_k = self.cache.get_seqlens_k(state.request_id, past_length, query_length)
 
+            # Then we update the total lengths that are used for slicing
             self.total_query_length += query_length
-            self.total_key_length += key_length
+            self.total_key_length += max(seqlens_k.values())  # this is used to slice the keys, so we need take the max
```
Comment (ArthurZucker) [inline src/transformers/generation/continuous_batching/continuous_api.py]:
```suggestion
            self.total_key_length += max(seqlens_k.values())  # this is used to slice the keys, so we need to take the max
```

File: src/transformers/generation/continuous_batching/continuous_api.py
Comment (ArthurZucker) [inline src/transformers/generation/continuous_batching/continuous_api.py:1]: On simplification is to always build a dict, but only when you create the `kwargs` do you slice if only 1 layer type

File: tests/generation/test_continuous_batching.py [hunk 3]
```diff
@@ -82,3 +87,204 @@ def test_group_layers(
...
+        )
+
+    @require_torch_gpu
+    @require_kernels
+    @slow
+    def test_continuous_batching_parity_gpt_oss_flash(self) -> None:
+        expected_outputs = {}
+        self._continuous_batching_parity(
+            "openai/gpt-oss-20b", "paged_attention|kernels-community/flash-attn", expected_outputs
+        )
```
Comment (ArthurZucker) [inline tests/generation/test_continuous_batching.py:286]: might make sense to use `fitures/cb_outputs.json` but up to you this is fine by me!

**Review by ArthurZucker — APPROVED** [review]
LGTM, supprised that sampling is not supported, but cudagraphs yes. Wondering if you just set slice_inputs = not cuda_graph ?

File: src/transformers/generation/continuous_batching/continuous_api.py (hunk 1 above)
Comment (remi-or) [reply src/transformers/generation/continuous_batching/continuous_api.py]: Right! Changed.

File: src/transformers/generation/continuous_batching/continuous_api.py (hunk 2 above)
Comment (remi-or) [reply src/transformers/generation/continuous_batching/continuous_api.py]: Changed

File: src/transformers/generation/continuous_batching/continuous_api.py
Comment (remi-or) [reply src/transformers/generation/continuous_batching/continuous_api.py:1]: Very true, mb. Changing to that.

**Comment by remi-or** [comment]
> supprised that sampling is not supported

I think there were issues last time, but I can check
> Wondering if you just set slice_inputs = not cuda_graph ?

Tried that it does not work for a few reasons, will look into restoring asap

File: tests/generation/test_continuous_batching.py (hunk 3 above)
Comment (remi-or) [reply tests/generation/test_continuous_batching.py:286]: We can do so in a next PR / when there is more tests as discussed offlie

File: src/transformers/generation/continuous_batching/requests.py [hunk 4]
```diff
@@ -25,7 +25,7 @@
 
 # We centralize the logger here to coordinate between logging and progress bar
 logger = logging.getLogger("ContinuousBatchingLogger")
-logger.setLevel(logging.INFO)
+# logger.setLevel(logging.INFO)
```
Comment (McPatate) [inline src/transformers/generation/continuous_batching/requests.py:28]: was this intentional btw?

File: src/transformers/utils/metrics.py
Comment (McPatate) [inline src/transformers/utils/metrics.py:1]: nice ty

File: src/transformers/generation/continuous_batching/requests.py (hunk 4 above)
Comment (remi-or) [reply src/transformers/generation/continuous_batching/requests.py:28]: Yes, seems like default should not be INFO. I can remove the comment next time, I agree it will be cleaner :)
